### PR TITLE
Release 0.1.0

### PR DESCRIPTION
**Pull request by cadic**
### Description of the Change

Prepares for the release 0.1.0
- Bump version
- Adds CHANGELOG.md
- Adds prepublishOnly script to build the package before publish
- Adds the GitHub Actions workflow to publish the NPM package on the release event

### How to test the Change
The `publish.yml` workflow was tested on the forked repo to my personal scope: https://www.npmjs.com/package/@cadic/cypress-wp-utils (will be deleted after the actual release)

### Checklist:

- [ ] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

**Comment by cadic** [comment]
@jeffpaul please make sure you've added `NPM_TOKEN` environment secret to the repo which will have publish access to the `@10up` organisation. It's also important to set the npmjs.com account 2FA not to "Require two-factor authentication for write actions" (otherwise the publish action will fail requiring the OTP)

Screenshot 2023-03-07 at 10 21 24

**Comment by cadic** [comment]
@jeffpaul The failing test against WordPress 6.2 is what we are working on #88 and hope to include to the 0.1.0 release, so I think it worth to wait with merging the current PR

**Comment by jeffpaul** [comment]
@cadic the `NPM_TOKEN` secret should now be available in the repo here

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  release:
+    types: [published]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    name: Publish
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (jeffpaul) [inline .github/workflows/publish.yml]: should we bump to v3 here as the latest major release branch?

File: package.json [hunk 2]
```diff
@@ -28,11 +29,7 @@
     "url": "git+https://github.com/10up/cypress-wp-utils.git"
   },
   "license": "MIT",
-  "author": {
-    "name": "10up",
-    "email": "10up@users.noreply.github.com",
-    "url": "https://github.com/10up"
-  },
+  "author": "10up <10up@users.noreply.github.com> (https://github.com/10up)",
```
Comment (jeffpaul) [inline package.json]: lets use `opensource@10up.com` for the email

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Changelog
+
+All notable changes to this project will be documented in this file, per the [Keep a Changelog standard](http://keepachangelog.com/), and will adhere to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased] - TBD
+
+## [0.1.0] - 2023-03-07
+- Initial release.
+
+[Unreleased]: https://github.com/10up/cypress-wp-utils/compare/trunk...develop
```
Comment (jeffpaul) [inline CHANGELOG.md:10]: we only have `trunk`, should we branch after 0.1.0 a `develop` branch and set that as the default branch?

**Review by jeffpaul — COMMENTED** [review]
some questions/comments that we should sort out while you look to get #88 through review/merged

**Review by jeffpaul — APPROVED** [review]
Looks good, thanks for working on the release!

**Comment by jeffpaul** [comment]
@cadic leaving this for you to merge & complete the tagging/releasing bits unless you'd prefer I handle that?

**Comment by cadic** [comment]
> @cadic leaving this for you to merge & complete the tagging/releasing bits unless you'd prefer I handle that?

@jeffpaul thanks for the review. I would prefer to tag and release myself after #89 will be reviewed, just want to include it into the release to make sure 0.1.0 will work with WordPress 6.2.

**Pull request merged**